internal/client.go

```diff
@@ -441,6 +446,15 @@ type (
 		CheckCallAllowed(ctx context.Context, method string, req, reply interface{}) error
 	}
 
+	// UpdateWorkflowExecutionOptions encapsulates the optional parameters for
+	// sending an update to a workflow execution.
+	// NOTE: Experimental
+	UpdateWorkflowExecutionOptions struct {
+		FirstExecutionRunID string
+		RunID               string
+		Header              *commonpb.Header
```

Comment:
In other calls, we keep the header hidden from non-interceptor API

Reply:
I took this from QueryWorkflowWithOptionsRequest which does have Header. But I'm not married to it if you'd prefer we hide it away.

Reply:
Lets leave it if we take the `UpdateWorkflowWithOptions` approach we took here with query